### PR TITLE
IGNITE-14144 Document C++ thin client transactions

### PR DESCRIPTION
**Pull request by Nikita-tech-writer**
Thank you for submitting the pull request to the Apache Ignite.

In order to streamline the review of the contribution 
we ask you to ensure the following steps have been taken:

### The Contribution Checklist
- [ ] There is a single JIRA ticket related to the pull request. 
- [ ] The web-link to the pull request is attached to the JIRA ticket.
- [ ] The JIRA ticket has the _Patch Available_ state.
- [ ] The pull request body describes changes that have been made. 
The description explains _WHAT_ and _WHY_ was made instead of _HOW_.
- [ ] The pull request title is treated as the final commit message. 
The following pattern must be used: `IGNITE-XXXX Change summary` where `XXXX` - number of JIRA issue.
- [ ] A reviewer has been mentioned through the JIRA comments 
(see [the Maintainers list](https://cwiki.apache.org/confluence/display/IGNITE/How+to+Contribute#HowtoContribute-ReviewProcessandMaintainers)) 
- [ ] The pull request has been checked by the Teamcity Bot and 
the `green visa` attached to the JIRA ticket (see [TC.Bot: Check PR](https://mtcga.gridgain.com/prs.html))

### Notes
- [How to Contribute](https://cwiki.apache.org/confluence/display/IGNITE/How+to+Contribute)
- [Coding abbreviation rules](https://cwiki.apache.org/confluence/display/IGNITE/Abbreviation+Rules)
- [Coding Guidelines](https://cwiki.apache.org/confluence/display/IGNITE/Coding+Guidelines)
- [Apache Ignite Teamcity Bot](https://cwiki.apache.org/confluence/display/IGNITE/Apache+Ignite+Teamcity+Bot)

If you need any help, please email dev@ignite.apache.org or ask anу advice on http://asf.slack.com _#ignite_ channel.

### CODE REVIEW

**Comment by Nikita-tech-writer** [comment]
@zstan Please take a look at the PR.

File: docs/_docs/thin-clients/cpp-thin-client.adoc [hunk 1]
```diff
@@ -115,3 +115,32 @@ Configure link:security/authentication[authentication on the cluster side] and p
 include::code-snippets/cpp/src/thin_authentication.cpp[tag=thin-authentication,indent=0]
 ----
 
+=== Executing Transactions
+
+The C++ thin client supports the transactional API. You can execute distributed transactions in various configuration
+modes by selecting required isolation levels and concurrency modes.
```
Comment (zstan) [inline docs/_docs/thin-clients/cpp-thin-client.adoc:121]: doubled "modes" seems excessive here.

**Comment by Nikita-tech-writer** [comment]
@zstan I've added more examples and aligned the topic with the similar one (thin client trasnactions) in java thin client. Please see the latest changes in the PR.

**Comment by zstan** [comment]
and what about :  transactions.withLabel(label) , we are talking previously ?

**Comment by Nikita-tech-writer** [comment]
I've added a paragraph on transactions with labels.

**Comment by Nikita-tech-writer** [comment]
@Mmuzaf Hi Maxim! Could you please cherry-pick this improvement to the 2.10 documentation?

File: docs/_docs/thin-clients/cpp-thin-client.adoc [hunk 2]
```diff
@@ -115,3 +115,75 @@ Configure link:security/authentication[authentication on the cluster side] and p
...
+transactions::ClientTransaction tx = transactions.withLabel(label).TxStart(TransactionConcurrency::OPTIMISTIC, TransactionIsolation::SERIALIZABLE, TX_TIMEOUT);
+----
+
+=== Init Container
+To use an init container, add the appropriate definitions.
+
+For example:
+----
+...
+init_container_image: busybox
```
Comment (zstan) [inline docs/_docs/thin-clients/cpp-thin-client.adoc:182]: why do we have container reference code here ?

**Comment by zstan** [comment]
@Nikita-tech-writer the one question: why do we have container reference changes here ? thanks !

File: docs/_docs/thin-clients/cpp-thin-client.adoc (hunk 2 above)
Comment (Nikita-tech-writer) [reply docs/_docs/thin-clients/cpp-thin-client.adoc]: It must have happened by mistake. Thanks for noticing!

**Comment by Nikita-tech-writer** [comment]
> @Nikita-tech-writer the one question: why do we have container reference changes here ? thanks !

Hi @zstan! It must have happened by mistake. Thanks for noticing! The issue is resolved, I've removed it.

**Comment by Nikita-tech-writer** [comment]
Hi @Mmuzaf, I do believe that this one can also be merged and cherry-picked to 2.10, if there is still a chance to do this. Thank you!

**Review by Mmuzaf — APPROVED** [review]

**Pull request merged**